### ISSUE TITLE
Project status

### ISSUE THREAD

**Issue by FredericGuilbault** [marcobergman/pypilot_conversion#1]
Hi, the main board of my st2000 died. and I found your project. Im more into programming than PCB, but I did a bit of stuff before. 

I was wondering what is the state of your project. Did it work ?  

Can I just take your Gerber files and send it right away to some manufacturer, or it's more of a work in progress ?

**Comment by marcobergman** [comment]
Yes, I have two of them on my boat and to much satisfaction; I'm sailing with this concept since 2019 and converted to these PCB's during covid. I presume you have seen the youtube videos.

There is an inherent drawback in the fact that the IMU is swung away from the tiller when not in use. Even though I lock the calibration when on standby, there seems to be some process that messes up some parameters which makes that right after hooking it up to the tiller again, it requires some time -less than a minute- to reset before you can use wind or gps mode. This is not a huge problem, but you need to be aware. The main drawback is that the ST2000 is far from water tight, so
you really need to be strict in closing it up properly, and maintaining the oil seal that's on the shaft. Still after some wet trips you need to open it up and dry it. A good water cover is recommended!

The build notes on the github readme mention some fault in the pcb design, but that can easily be tweaked as indicated.

Any questions, feel free to post them in this issue, so others may benefit!

Cheers,

/Marco Bergman

**Comment by FredericGuilbault** [comment]
Thx , I will also try summit few pull request to make it more noob friendly as I'm the perfect use case of noob experience :)

**Comment by FredericGuilbault** [comment]
in the README file there is this sentence `Solved by cutting the pin 22 off the arduino and putting a resistor between arduino pins 22 and 27.` I see the resistor on one of the pictures, but I can't read its value. What is it ?

**Comment by marcobergman** [comment]
It appears to be 12k. Anything between 100k and 1k will do in this case. I think I had it lying on my desk. Even a(n insulated) wire would do but the insulation tends to melt and become messy.

**Comment by FredericGuilbault** [comment]
So, just to be sure, no resistance is needed, I can simply do like this ? 

![27 to 22](https://github.com/marcobergman/pypilot_conversion/assets/37122999/d26a354f-051f-44d2-be1c-1f4e3155d5c0)

**Comment by marcobergman** [comment]
Yes, I'm pretty sure you can.

**Comment by FredericGuilbault** [comment]
I don't find the buzzer responding to the dimensions of the bill of material `D14mm height 7mm with pitch 10mm`
All i find is either 17MM in diameter, or the pitch is in fact 7.6MM. do you know the part number of what you use or is it possible that the physical diameter is 17MM, and only the fingerprint is 14MM ? 

Thx again for your time.

**Comment by FredericGuilbault** [comment]
also, I can`t find the value of the resistor R2 for the buzzer. do you remember ?

![image](https://github.com/marcobergman/pypilot_conversion/assets/37122999/9302f3fc-d73c-4674-99f6-1fb7d1eb9668)

**Comment by marcobergman** [comment]
> I don't find the buzzer responding to the dimensions of the bill of material `D14mm height 7mm with pitch 10mm` All i find is either 17MM in diameter, or the pitch is in fact 7.6MM. do you know the part number of what you use or is it possible that the physical diameter is 17MM, and only the fingerprint is 14MM ?
> 
> Thx again for your time.

**Comment by marcobergman** [comment]
I think I used a buzzer from an automotive store, the type that is sold to warn when you left your light on. Took it out of its casing. For the resistor I'd use a 1k or 560ohm one.

**Comment by FredericGuilbault** [comment]
I would also some help to understand the current sensing from pololu. 

For the CS pin, their website it says: 
`Current sense output: This pin outputs a voltage proportional to the motor current when the H-bridge is driving (but not while it is braking, including when current limiting is active). The output voltage is about 20 mV/A plus a 50 mV offset.`
So as i understand it, we are in a context where more volts means more amperage. 

but on the  [pypilot](https://github.com/pypilot/pypilot/tree/master)/[arduino](https://github.com/pypilot/pypilot/tree/master/arduino)/[motor](https://github.com/pypilot/pypilot/tree/master/arduino/motor/motor.ino ) file at line 14 they are expecting a shunt. Therefor i imagine a voltage drop as a mesure of amperage. 

And there is the R9 resistance of 560hms on the plan in between. 

I started digging that part cuz im having `OVERCURRENT_FAULT` messages from `pypilot_servo` as soon as it boot, even before engaging the autopilot :
```
voltage: 12.4 current 10.08 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
voltage: 12.4 current 10.78 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
voltage: 12.39 current 9.43 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
voltage: 12.4 current 9.67 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
voltage: 12.4 current 9.97 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
voltage: 12.41 current 9.19 ctrl temp 11.74 motor temp False rudder pos False flags SYNC OVERCURRENT_FAULT 
```

Do you remember modifing something in the code about that ?

**Comment by marcobergman** [comment]
According to the (2-page!) schematics https://pypilot.org/schematics/hbridge_controller.pdf, the shunt is in the negative side, meaning the voltage goes up as the current goes up, which makes the pololu output theoretically compatible with the software. On the second page you'll find the seemingly non-functional 560ohm resistor towards the A1, I just copied it from this original design out of superstition I think.

Myself, I never got the current sensing entirely linear, but it is a bit hard to calibrate; I can only measure during the extension of the rod which is only a few seconds, during which my electronic multimeter shows many different readings. 

I don't have one of my operational pypilot readily available at the moment, but for my calibration attempts I remember to have used these 2 parameters https://github.com/pypilot/workbook/wiki/parameters#:~:text=fluctuating%20battery%20voltage%20(!).-,servo.current.factor,-default%3A%201%3B%20screen. Hope this helps.

Edit: I might have tweaked motor.ino a bit, as I can see the following diff in what appears my latest ino:

```
887c906
<                 v = v  * 625 / 128 / 16;
---
>                 v = v  * 525 / 128 / 16;
```

The fact that I did not document this properly, probably means that I never saw any benefit of this tweak.

In the end, I decided not to care about proper current sensing, as I have the end-of-travel switches anyway.

Edit 2: In your case, getting a reading of > 8A in rest, I'd check with a volt meter what reading you have on A1. Ultimately you could set servo.max_current to a high limit. Myself, I seem to have 18A there, which i way more than could practically run through these wires. The forementioned servo.current. parameters don't seem to work for me, but I have an odd mix of old software versions in use.
